openupgrade_scripts/scripts/hr/15.0.1.1/post-migration.py

```diff
@@ -0,0 +1,60 @@
+from openupgradelib import openupgrade
+
+
+def compute_work_location_id(env):
+    env.cr.execute("SELECT * FROM hr_employee WHERE work_location IS NOT NULL")
+    all_employees = env.cr.dictfetchall()
+
+    work_location_dict = {}
+    for employee in all_employees:
+        if employee["work_location"] not in work_location_dict.keys():
+            address_id = employee.get("address_id", False)
+            if not address_id:
+                address_id = (
+                    env["res.partner"]
+                    .create(
+                        {
+                            "name": employee["work_location"],
+                            "company_type": "person",
+                            "type": "private",
+                        }
+                    )
+                    .id
+                )
+            work_location = env["hr.work.location"].create(
+                {
+                    "name": employee["work_location"],
+                    "company_id": employee["company_id"],
+                    "address_id": address_id,
+                }
+            )
+            work_location_dict[employee["work_location"]] = work_location
+        env["hr.employee"].browse(employee["id"]).work_location_id = work_location_dict[
+            employee["work_location"]
+        ]
```

Comment:
@daonamutc1 anh test giúp em đoạn này nhé
```suggestion
def _map_hr_employee_work_location_to_work_location_id(env):
    env.cr.execute("""
        WITH work_location_tmp AS (
            INSERT INTO hr_work_location (name, company_id, address_id)
            SELECT emp.work_location, emp.company_id, emp.address_id
            FROM hr_employee emp
            WHERE emp.address_id IS NOT NULL AND emp.work_location IS NOT NULL
            RETURNING id, emp.id as emp_id
        )
        UPDATE hr_employee emp
        SET work_location_id = work_location_tmp.id
        FROM work_location_tmp
        WHERE work_location_tmp.emp_id = emp.id
```